akka-persistence-typed/src/main/scala/akka/persistence/typed/dstate/DurableStateSignal.scala

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.typed.dstate
+
+import akka.actor.typed.Signal
+import akka.annotation.DoNotInherit
+
+/**
+ * Supertype for all Akka Persistence Typed - Durable State specific signals
+ *
+ * Not for user extension
+ */
+@DoNotInherit
+sealed trait DurableStateSignal  extends Signal
+
+@DoNotInherit sealed abstract class RecoveryCompleted extends DurableStateSignal
+case object RecoveryCompleted extends RecoveryCompleted {
+  def instance: RecoveryCompleted = this
+}
+
+final case class RecoveryFailed(failure: Throwable) extends DurableStateSignal {
+
+  /**
+   * Java API
+   */
+  def getFailure(): Throwable = failure
+}
+
+
+case object PersistenceCompleted extends DurableStateSignal {
+
+  /**
+   * Java API
+   */
+  def instance: PersistenceCompleted.type = this
+}
+
+final case class PersistenceFailed(failure: Throwable) extends DurableStateSignal {
+
+  /**
+   * Java API
+   */
+  def getFailure(): Throwable = failure
+
+}
+
+
+case object DeletionCompleted extends DurableStateSignal {
+
+  /**
+   * Java API
+   */
+  def instance: DeletionCompleted.type = this
+}
+
+
+final case class DeletionFailed(failure: Throwable) extends DurableStateSignal {
+
+  /**
+   * Java API
+   */
+  def getFailure(): Throwable = failure
+
+}
```

Comment:
I wonder about how useful these signals are vs stopping the actor when deleted and crashing it when persisting fails - can it really be ok with the actor really keep running after those events?

Reply:
I don't know what is the best to do here. Should we preserve the mailbox? 

In any case, if we delete, we should treat it as a new incarnation. Meaning, we should go back to `initialState`. If we let it crash, it will happen automatic. If we want to preserve the mailbox, we need to restart it but move to `initialState`. 

For general persistence failures, I don't think we should let it crash if we can guarantee that the state was not mutated. And that's guaranteed.

Reply:
For persist failures we should treat them in the same way as for EventSourcedBehavior. It's stopping. The reason is that it's typically not possible to know if it was stored or not in the database (e.g. timeouts) and therefore it must be loaded again.